### ISSUE TITLE
Masson2 api discuss

### ISSUE THREAD

**Issue by lucasbak** [adaltas/node-masson#33]
as agreed we should discuss further @wdavidw [proposition](https://github.com/adaltas/node-masson/commit/1b447c0565a04be68c2324c24217297f9e3859e1), [here](https://github.com/adaltas/node-masson/commit/b4692b81532fa13ceb8a5154386e4a7cc0f9b125) is mine

**Comment by leopaul36** [comment]
I agree with @lucasbak 's proposition to rename `auto: true` to `install: true`, I think it's more understandable.
Is `single: true` really useful ? Can't we just use `[0]` to avoid confusion with `local: true` ? Just a thought...
@wdavidw I think `recommended: true` would just bring confusion. A dependency is either mandatory or not, isn't it ?
Also are there use cases for `min, max` ?
